### ISSUE TITLE
Failed to initialize vscode-spotify. We'll keep trying every 20 seconds.

### ISSUE THREAD

**Issue by clemtoy** [ShyykoSerhiy/vscode-spotify#13]
Hi,
The extension give this error message: `Failed to initialize vscode-spotify. We'll keep trying every 20 seconds.`

Spotify is running.

![image](https://cloud.githubusercontent.com/assets/5541136/24300999/983ac4ca-10ae-11e7-8bd7-426bcafad2d1.png)

(A proxy is set in both spotify and vscode configuration.)

Thank you in advance for your help.

**Comment by ShyykoSerhiy** [comment]
Hi, @clemtoy, thx for bug report. It seems like in your case SpotifyWebHelper spawned on port 4381. Usually It stays in 4370 - 4380 range. I'll bump the range to 4370-4390 when trying to connect and add config params to change it manually for extension. The fix will be ready in 2-3 hours.

**Comment by ShyykoSerhiy** [comment]
@clemtoy , meanwhile can you please try to change 
C:\Users\cmichard\.vscode\extensions\shyykoserhiy.vscode-spotify-1.1.0\node_modules\spotilocal\dist\src\index.js on code line 121 
```ts
if (port > 4380)
```
to 
```ts
if (port > 4390)
```
and check if it works? You'll need to restart vscode in order to try it. Thx

**Comment by clemtoy** [comment]
@ShyykoSerhiy 
It does not work :

![image](https://cloud.githubusercontent.com/assets/5541136/24308312/c8de6d0c-10c7-11e7-9a13-693213ab686c.png)

Moreover, the error message is not "It looks like spotify isn't open [...]"

**Comment by ShyykoSerhiy** [comment]
Oh, my bad) I've missed the part 

> (A proxy is set in both spotify and vscode configuration.)

So you've set "http.proxy" in vscode settings, right?

**Comment by ShyykoSerhiy** [comment]
Yeah, I don't use proxy settings provided by vscode in extension. I'll fix this by using the proxy if "https.proxy" and/or "http.proxyStrictSSL" settings are provided. 
That's probably what causes errors for you. 
AND I need to change https://github.com/ShyykoSerhiy/vscode-spotify/blob/master/src/spotify/OsAgnosticSpotifyClient.ts#L64 error to provide more relevant info.  
Thx.

**Comment by clemtoy** [comment]
Yes! Thank you in advance.

**Comment by clemtoy** [comment]
Another remark: you use a stop icon instead of pause. It is a bit confusing.
Great job any way

**Comment by iTouchTheSky** [comment]
Was wondering if I should open another bug, but I currently have this message with the latest VSCode Insiders version it wont find my spotify.

**Comment by ShyykoSerhiy** [comment]
@iTouchTheSky, have i been working before?

**Comment by ShyykoSerhiy** [comment]
@iTouchTheSky  you probably should open another bug

**Comment by bernardomed** [comment]
I'm getting the same error on my iMac running Sierra.

**Comment by ShyykoSerhiy** [comment]
@bernardomed what version of vscode-spotify do you have? 
Have you set "spotify.useCombinedApproachOnMacOS" param to : true?

**Comment by bernardomed** [comment]
@ShyykoSerhiy I'm running vscode-spotify 2.1.1. No, I haven't.

**Comment by bernardomed** [comment]
@ShyykoSerhiy Adding `"spotify.useCombinedApproachOnMacOS": true` fixed my issue.

**Comment by nilshartmann** [comment]
For me, setting `"spotify.useCombinedApproachOnMacOS": true` does not solve the problem. Spotify is running before I launch VS Code.

BTW: The Commands (like play and pause) are working, but there is nothing displayed in the status bar (no buttons, no title, ...)

MacOS Sierra (10.12.6)
VSCode Version 1.17.1
Spotify: 1.0.65.320.gac7a8e02
vscode-spotify: 2.2.1
Excerpt from my User Settings:
```
  "spotify.showVolumeUpButton": true,
  "spotify.showVolumeDownButton": true,
  "spotify.showToggleShufflingButton": true,
  "spotify.useCombinedApproachOnMacOS": true,
  "spotify.showInitializationError": true,
"http.proxy": "http://localhost:3128"
```
Same Proxy configured in Spotify.

Any Ideas? Thanks!

**Comment by ShyykoSerhiy** [comment]
@nilshartmann, try "spotify.useCombinedApproachOnMacOS": false instead. Error with proxy will be fixed in next realse

**Comment by nilshartmann** [comment]
@ShyykoSerhiy sorry for my late response. Setting "spotify.useCombinedApproachOnMacOS" to "false" was the trick. Now it works. Yeah! Thanks! 🎸

**Comment by ShyykoSerhiy** [comment]
Connected to https://github.com/Microsoft/vscode/issues/12588
